woocommerce/woocommerce
# Add more specific reason for why products cannot be purchased so plug…

### All Submissions:






### Changes proposed in this Pull Request:

Provide more information on why an item cannot be purchased and provide a filter so plugins can modify the error message however they want. 

Copy changes are welcome, I used what made the most sense to me but maybe there's something better. In particular I wasn't sure what to say when a product in unpublished. Went withhttps://github.com/woocommerce/woocommerce/compare/fix/issue-23750?expand=1#diff-bad16647d10a7365519084a2fc793cbbR1519

Closes #23750.

### How to test the changes in this Pull Request:

1. Load an item's product page.
2. Update that item's price to '' and save it in the admin.
3. Without reloading the item's product page, add the item to the cart.
4. Error notice of "Sorry, this product cannot be purchased because it has no price set."
Screenshot of this in action.
![Screenshot from 2019-07-16 12-53-24](https://user-images.githubusercontent.com/1534605/61335184-d4bebf80-a7e9-11e9-941c-338ef53196f0.png)


1. Add an item to the cart.
2. Update that item's price to '' and save it in the admin.
3. Reload cart page.
4. Two error notices should appear. "product_name has been removed from your cart because it can no longer be purchased. Please contact us if you need assistance." and "Sorry, this product cannot be purchased because it has no price set."

### Other information:




### Changelog entry

Use more descriptive error when a product is not purchasable. Add `woocommerce_is_purchasable_message` to customize these messages.
